pages/scenario_selection_page.py

```diff
@@ -309,67 +381,344 @@ def scenario_selection_page():
+        "lifetime_running_costs": "Running costs",
+
+    }
 
-    csv_annual = dummy_data.to_csv()
-    csv_total = total_dummy_data.to_csv()
+    df_comparing_costs["cost_type"] = df_comparing_costs["cost_type"].map(cost_type_name_mapping)
+
+    # Convert column to ordered categorical
+    df_comparing_costs["archetype_label"] = pd.Categorical(
+        df_comparing_costs["archetype_label"],
+        categories=mapped_archetype_options + ["Weighted average archetype"],
+        ordered=True
+    )
+    df_comparing_costs["cost_type"] = pd.Categorical(
+        df_comparing_costs["cost_type"],
+        categories=list(cost_type_name_mapping.values()),
+        ordered=True
+    )
+    df_comparing_costs = df_comparing_costs.sort_values(by=["archetype_label", "cost"])
+
+    cost_component_order = {
+        "Loan interest":0,
+        "Maintenance costs":1,
+        "Installation costs (after subsidy)":2,
+        "Running costs":3,
+    }
+    df_comparing_costs["cost_type_order"] = df_comparing_costs["cost_type"].map(cost_component_order)
+
+    chart = (
+        alt.Chart(df_comparing_costs)
+        .mark_bar()
+        .encode(
+            y=alt.Y("technology:N", axis=alt.Axis(title=None)),
+            x=alt.X("sum(cost):Q", axis=alt.Axis(title="£")),
+            color=alt.Color(
+                "cost_type:N",
+                title="Cost component",
+                scale=alt.Scale(
+                    domain=cost_type_name_mapping.values(),
+                    range=[
+                        NESTA_COLOURS[10],
+                        NESTA_COLOURS[2],
+                        NESTA_COLOURS[1],
+                        NESTA_COLOURS[0],
+                    ],
+                ),
+            ),
+            order=alt.Order(
+                "cost_type_order:N",
+                sort="descending"
+            )
+        )
+        .facet(
+            row=alt.Row(
+                "archetype_label:N", 
+                header=alt.Header(labelAngle=0, labelAlign="left"),
+                title=None,
+            )
+        )
+    )
+
+    st.markdown("You can toggle between annualised lifetime costs and total lifetime costs below. By default, annualised lifetime costs are shown, i.e. total lifetime costs divided by the lifespan of the heating system.")
+
+    toggle_total_lifetime_costs = st.toggle(
+        label="Show total lifetime costs instead of annualised lifetime costs",
+        value=False,
+        key="toggle_total_lifetime_costs",
+        help="Toggle between annualised lifetime costs and total lifetime costs",
+    )
+
+    if toggle_total_lifetime_costs:
+        title = ["Total lifetime costs of heating systems, broken down by installation costs (after subsidy),",
+        " running costs, maintenance costs and loan interest (if applicable)"]
+    else:
+        df_comparing_costs["cost"] = (
+            df_comparing_costs["cost"] / df_comparing_costs["life_span"]
+        )
```

Comment:
If we do implement the total_cost column, need to divide this by lifespan too

```suggestion
        df_comparing_costs["cost"] = (
            df_comparing_costs["cost"] / df_comparing_costs["life_span"]
        )
        df_comparing_costs["total_cost"] = (
            df_comparing_costs["total_cost"] / df_comparing_costs["life_span"]
        )
```